libs/vm-modules/tests/unit/ml/ml_model_tests.cpp

```diff
@@ -847,4 +852,60 @@ TEST_F(VMModelTests, model_with_accuracy_metric)
   EXPECT_LE(metrics->elements.at(1), 1);
 }
 
+TEST_F(VMModelTests, DISABLED_model_sequential_flatten)
+{
+  static char const *SRC_METRIC = R"(
+        function main()
+          var model = Model("sequential");
+          model.add("flatten");
+          model.compile("scel", "adam", {"categorical accuracy"});
+        endfunction
+      )";
+
+  ASSERT_TRUE(toolkit.Compile(SRC_METRIC));
+  ASSERT_TRUE(toolkit.Run());
+}
+
+TEST_F(VMModelTests, DISABLED_model_sequential_flatten_tensor_data)
+{
+  static char const *SRC_METRIC = R"(
+        function main() : Tensor
+          var tensor_shape = Array<UInt64>(2);
+          tensor_shape[0] = 3u64;
+          tensor_shape[1] = 3u64;
+
+          var x = Tensor(tensor_shape);
+          var str_vals = "0.5, 7.1, 9.1; 6.2, 7.1, 4.; -99.1, 14328.1, 10.0;";
```

Comment:
as we discussed, above creation could be reworked just to
var x = Tensor(Array<UInt64>(1));
because later it is resized and reshaped anyway.